### PR TITLE
(feature, typescript): Add setObjectProperty core utility

### PR DESCRIPTION
**Pull request by amckinney**
This adds a new `setObjectProperty` core utility which includes the `setObjectProperty` function. This function acts similar to [Lodash.set](https://lodash.com/docs/4.17.15#set), but is rewritten and simplified to accommodate our use case without bringing in a large dependency.

This function is useful for setting arbitrarily nested values on objects within the SDK generator like so:

```typescript
const person = {
  age: 42
}
const john = setObjectProperty(person, "name", "John Doe");
```

Note that this fixes an issue for pagination endpoints that define nested offset/cursor properties. The generated pagination code now looks like the following:

```typescript
let _offset = request?.pagination?.page != null ? request?.pagination?.page : 1;
return new core.Pageable<SeedPagination.ListUsersPaginationResponse, SeedPagination.User>({
  response: await list(request),
  hasNextPage: (response) => (response?.data ?? []).length > 0,
  getItems: (response) => response?.data ?? [],
  loadPage: (_response) => {
    _offset += 1;
    return list(core.setObjectProperty(request, "pagination.page", _offset));
  }
});
```

### CODE REVIEW

File: generators/typescript/utils/core-utilities/reflect/__test__/setObjectProperty.test.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+interface TestCase {
+    description: string;
+    giveObject: object;
+    givePath: string;
+    giveValue: any;
+    wantObject: object;
+}
+
+describe("Test setObjectProperty", () => {
```
Comment (dsinghvi) [inline generators/typescript/utils/core-utilities/reflect/__test__/setObjectProperty.test.ts:11]: @amckinney i think `reflect` is probably a little weird in TS (not going to be easy for someone poking in to understand. 

Do we have a `utils` folder in core? I could see this just going in there as one of the helper functions ? or a `helpers` folder?

File: generators/typescript/utils/core-utilities/reflect/setObjectProperty.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+/**
+ * Sets the value at path of object. If a portion of path doesn’t exist it’s created.
+ *
+ * @param object The object to modify.
+ * @param path The path of the property to set.
+ * @param value The value to set.
+ * @return Returns object.
+ */
+export function setObjectProperty<T extends object>(object: T, path: string, value: any): T {
```
Comment (dsinghvi) [inline generators/typescript/utils/core-utilities/reflect/setObjectProperty.ts:9]: lets credit lodash if we took it from there ?

Comment (amckinney) [reply generators/typescript/utils/core-utilities/reflect/setObjectProperty.ts:9]: Oops, totally meant to do that - will add that in.

File: generators/typescript/utils/core-utilities/reflect/__test__/setObjectProperty.test.ts (hunk 1 above)
Comment (amckinney) [reply generators/typescript/utils/core-utilities/reflect/__test__/setObjectProperty.test.ts:11]: Sounds good, we don't have a good place for it already, but happy to name it something more generic like `utils`.

**Review by dsinghvi — APPROVED** [review]

**Pull request merged**